packages/core/src/stateUtils.ts

```diff
@@ -1035,11 +936,26 @@ function getEffectiveTargetStates<TC, TE extends EventObject>(
   transition: TransitionDefinition<TC, TE>,
   state: State<TC, TE>
 ): Array<StateNode<TC, any, TE>> {
+  if (!transition.target) {
+    return [];
+  }
+
   const targets = new Set<StateNode<TC, any, TE>>();
 
-  for (const s of transition.target || []) {
+  for (const s of transition.target) {
     if (isHistoryNode(s)) {
-      resolveHistory(s, state).forEach(t => targets.add(t));
+      if (state.historyValue[s.id]) {
+        state.historyValue[s.id].forEach(node => {
+          targets.add(node);
+        });
+      } else {
+        getEffectiveTargetStates(
+          { target: resolveHistoryTarget(s) } as TransitionDefinition<TC, TE>,
```

Comment:
this is also a hack, as the only thing actually needed by `getEffectiveTargetStates` is the target property, but this will be refactored once we add a default transition to history states, that can be done in a followup PR after both this PR and the microstep work lands on the next branch